File: autostart.py
from os import system
from datetime import datetime
from settings.path import qtile_path, picom_path
import subprocess


def night(current_hour : int):
    if current_hour in range(17, 24) or current_hour in range(0,10):
        system('redshift -x')
        system('redshift -O 4000')
    if current_hour in range(19, 24) or current_hour in range(0,12): 
        system('brightnessctl set 30%')

def day(current_hour : int):
    if current_hour in range(13, 16): system('redshift -x')
    if current_hour in range(11, 18): system('brightnessctl set 80%')

def run():
    system(f'~/bin/eww daemon')
   # cmd = subprocess.Popen(f'~/bin/eww daemon', stdout=subprocess.PIPE, shell=True)
   # output, err = cmd.communicate()
   # p_status = cmd.wait()
    system(f'picom &')
    system(f'udiskie &')
    system(f'~/.fehbg &')
    system(f'dunst &')
    system(f'playerctld daemon &')
    #system(f'bash {qtile_path}/scripts/feh-blur &')
    
if __name__ == '__main__':
    current_hour : int = int(datetime.now().strftime("%H"))
    night(current_hour)
    day(current_hour)
    run()
    

File: autostart.go
/*
compile after making changes
*/

package main

import (
	"fmt"
	"os/exec"
	"time"
)

type Exec struct {
	command  string
	argument []string
}

func get_time(hour, min *int) {
	*hour = time.Now().Hour()
	*min = time.Now().Minute()
}

func (cmd Exec) executor() {
	for _, e := range cmd.argument {
		err := exec.Command(cmd.command, e).Run()
		if err != nil {
			fmt.Println(err)
		}
	}
}

func main() {

	eww_daemon := Exec{
		command:  "~/bin/eww",
		argument: []string{"daemon"},
	}
	picom := Exec{
		command:  "picom",
		argument: []string{"&"},
	}
	udiskie := Exec{
		command:  "udiskie",
		argument: []string{"&"},
	}
	feh := Exec{
		command:  "~/.fehbg",
		argument: []string{"&"},
	}
	dunst := Exec{
		command:  "dunst",
		argument: []string{"&"},
	}

	var current_hour, current_minute int
	get_time(&current_hour, &current_minute)

	if current_hour >= 17 && current_minute >= 0 {
		redshift := Exec{
			command:  "redshift",
			argument: []string{"-C", "4000"},
		}
		brightness := Exec{
			command:  "brightnessctl",
			argument: []string{"set", "30%"},
		}

		redshift.executor()
		brightness.executor()

	} else {
		redshift := Exec{
			command:  "redshift",
			argument: []string{"-x"},
		}
		brightness := Exec{
			command:  "brightnessctl",
			argument: []string{"set", "80%"},
		}

		redshift.executor()
		brightness.executor()
	}

	eww_daemon.executor()
	picom.executor()
	udiskie.executor()
	feh.executor()
	dunst.executor()

}
